### ISSUE TITLE
Error running test_simple on refactor-rngs-7

### ISSUE THREAD

**Issue by robynstuart** [starsimhub/starsim#410]
Again, maybe due to Python 3.9. Happens during `test_demo`

```
Traceback (most recent call last):
  File "/Users/robynstuart/anaconda3/envs/starsim/lib/python3.9/site-packages/sciris/sc_odict.py", line 1160, in __getattribute__
    return odict.__getattribute__(self, attr)
RecursionError: maximum recursion depth exceeded while calling a Python object
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/Users/robynstuart/anaconda3/envs/starsim/lib/python3.9/site-packages/IPython/core/interactiveshell.py", line 3526, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-2-05dab0875927>", line 1, in <module>
    runfile('/Users/robynstuart/Documents/git/starsim/tests/test_simple.py', wdir='/Users/robynstuart/Documents/git/starsim/tests')
  File "/Applications/PyCharm.app/Contents/plugins/python/helpers/pydev/_pydev_bundle/pydev_umd.py", line 198, in runfile
    pydev_imports.execfile(filename, global_vars, local_vars)  # execute the script
  File "/Applications/PyCharm.app/Contents/plugins/python/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/Users/robynstuart/Documents/git/starsim/tests/test_simple.py", line 155, in <module>
    s0 = test_demo()
  File "/Users/robynstuart/Documents/git/starsim/tests/test_simple.py", line 18, in test_demo
    sim = ss.demo()
  File "/Users/robynstuart/Documents/git/starsim/starsim/sim.py", line 975, in demo
    sim.run()
  File "/Users/robynstuart/Documents/git/starsim/starsim/sim.py", line 526, in run
    self.initialize()
  File "/Users/robynstuart/Documents/git/starsim/starsim/sim.py", line 97, in initialize
    self.dists.initialize(obj=self, base_seed=self.pars.rand_seed + 2)  # +2 ensures that seeds from the above population initialization and the +1-offset below are not reused within the rngs
  File "/Users/robynstuart/Documents/git/starsim/starsim/distributions.py", line 79, in initialize
    self.dists = find_dists(obj)
  File "/Users/robynstuart/Documents/git/starsim/starsim/distributions.py", line 43, in find_dists
    tree = sc.iterobj(obj) # Temporary copy of sc.iterobj(obj) until Sciris 3.1.5 is released
  File "/Users/robynstuart/anaconda3/envs/starsim/lib/python3.9/site-packages/sciris/sc_nested.py", line 459, in iterobj
    out = io.iterate() # Iterate
  File "/Users/robynstuart/anaconda3/envs/starsim/lib/python3.9/site-packages/sciris/sc_nested.py", line 388, in iterate
    io.iterate() # Run recursively
  File "/Users/robynstuart/anaconda3/envs/starsim/lib/python3.9/site-packages/sciris/sc_nested.py", line 388, in iterate
    io.iterate() # Run recursively
  File "/Users/robynstuart/anaconda3/envs/starsim/lib/python3.9/site-packages/sciris/sc_nested.py", line 388, in iterate
    io.iterate() # Run recursively
  [Previous line repeated 2963 more times]
  File "/Users/robynstuart/anaconda3/envs/starsim/lib/python3.9/site-packages/sciris/sc_nested.py", line 384, in iterate
    io = IterObj(self.getitem(key), self.func, inplace=self.inplace, leaf=self.leaf,  # Create a new instance
  File "/Users/robynstuart/anaconda3/envs/starsim/lib/python3.9/site-packages/sciris/sc_nested.py", line 315, in __init__
    self.itertype = self.check_iter_type(self.obj)
  File "/Users/robynstuart/anaconda3/envs/starsim/lib/python3.9/site-packages/sciris/sc_nested.py", line 367, in check_iter_type
    return check_iter_type(obj, known=self.atomic, custom=self.custom_type)
  File "/Users/robynstuart/anaconda3/envs/starsim/lib/python3.9/site-packages/sciris/sc_nested.py", line 110, in check_iter_type
    if known is not None and isinstance(obj, known):
  File "/Users/robynstuart/anaconda3/envs/starsim/lib/python3.9/site-packages/sciris/sc_odict.py", line 1162, in __getattribute__
    return self.__getitem__(attr, exception=E)
RecursionError: maximum recursion depth exceeded
```

**Comment by cliffckerr** [comment]
@robynstuart I think you need to update Sciris, see https://github.com/starsimhub/starsim/issues/413

**Comment by cliffckerr** [comment]
Closed by #420
